src/DrupalCI/Configuration/ConfigurationManager.php

```diff
@@ -0,0 +1,98 @@
+<?php
+
+namespace DrupalCI\Configuration;
+
+use Doctrine\Common\Annotations\AnnotationReader;
+use Doctrine\Common\Annotations\AnnotationRegistry;
+use Drupal\Component\Annotation\Variable;
+use Drupal\Component\Configuration\Exception\EnvironmentVariableNotSetException;
+
+class ConfigurationManager {
+
+  /**
+   * A list of all environment variables.
+   *
+   * @var array
+   */
+  protected $environmentVariables = [];
+
+  /**
+   * A list of defined variables.
+   *
+   * @var array
+   */
+  protected $variables;
+
+  /**
+   * Constructs a ConfigurationManager.
+   */
+  public function __construct() {
+    AnnotationRegistry::registerLoader('class_exists');
+    $this->discoverVariables();
+    $this->parseEnvironmentVariables();
+  }
+
+  public function test() {
+    $this->discoverVariables();
+  }
+
+  /**
+   * Load all the environment variables.
+   */
+  private function parseEnvironmentVariables() {
+    $environment_stacks = [$_ENV, $_SERVER];
+    foreach ($environment_stacks as $environment_stack) {
+      foreach ($environment_stack as $item => $value) {
+        if (substr($item, 0, 4) == "DCI_") {
+          $this->environmentVariables[$item] = $value;
+        }
+      }
+    }
+  }
+
+  /**
+   * Discovers the list of defined variables.
+   */
+  protected function discoverVariables() {
+    $reflectionClassName = 'DrupalCI\\Configuration\\EnvironmentVariables';
+    $reader = new AnnotationReader();
+    $reflectionClass = new \ReflectionClass($reflectionClassName);
+
```

Comment:
And once The properties are defined on the plugin class, we need a way to iterate over all the BuildTasks in the system, so we can getPropertyAnnotations on each of them.